pytorch/workflows/mnist.py

```diff
@@ -0,0 +1,181 @@
+
+    if should_distribute():
+        print('Using distributed PyTorch with {} backend'.format(backend_type))
+        dist.init_process_group(backend=backend_type)
+
+    kwargs = {'num_workers': 1, 'pin_memory': True} if use_cuda else {}
+    train_loader = torch.utils.data.DataLoader(
+        datasets.MNIST('../data', train=True, download=True,
+                       transform=transforms.Compose([
+                           transforms.ToTensor(),
+                           transforms.Normalize((0.1307,), (0.3081,))
+                       ])),
+        batch_size=batch_size, shuffle=True, **kwargs)
+    test_loader = torch.utils.data.DataLoader(
+        datasets.MNIST('../data', train=False, transform=transforms.Compose([
+            transforms.ToTensor(),
+            transforms.Normalize((0.1307,), (0.3081,))
+        ])),
+        batch_size=test_batch_size, shuffle=False, **kwargs)
+
+    model = Net().to(device)
+
+    if is_distributed():
+        Distributor = nn.parallel.DistributedDataParallel if use_cuda and torch.cuda.is_available \
+            else nn.parallel.DistributedDataParallelCPU
+        model = Distributor(model)
+
+    optimizer = optim.SGD(model.parameters(), lr=learning_rate, momentum=sgd_momentum)
+
+    accuracies = [epoch_step(model, device, train_loader, test_loader, optimizer, epoch, writer, log_interval) for epoch in range(1, epochs + 1)]
+
+    if (save_model):
+        torch.save(model.state_dict(),"mnist_cnn.pt")
+
+    out.set(str(accuracies))
+
+
+@workflow_class
+class MNISTTest(object):
+    no_cuda = Input(Types.Boolean, default=False, help="disables CUDA training")
+    batch_size = Input(Types.Integer, default=64, help='input batch size for training (default: 64)')
+    test_batch_size = Input(Types.Integer, default=1000, help='input batch size for testing (default: 1000)')
+    epochs = Input(Types.Integer, default=1, help='number of epochs to train (default: 10)')
+    learning_rate = Input(Types.Float, default=0.01, help='learning rate (default: 0.01)')
+    sgd_momentum = Input(Types.Float, default=0.5, help='SGD momentum (default: 0.5)')
+    seed = Input(Types.Integer, default=1, help='random seed (default: 1)')
+    log_interval = Input(Types.Integer, default=10, help='how many batches to wait before logging training status')
+    save_model = Input(Types.Boolean, default=False, help='For Saving the current Model')
+    dir = Input(Types.String, default='logs', help='directory where summary logs are stored')
+
+    mnist_result = mnist_pytorch_job(
+        no_cuda=no_cuda,
+        batch_size=batch_size,
+        test_batch_size=test_batch_size,
+        epochs=epochs,
+        learning_rate=learning_rate,
+        sgd_momentum=sgd_momentum,
+        seed=seed,
+        log_interval=log_interval,
+        save_model=save_model,
+        dir=dir
+    )
+
+    accuracies = Output(mnist_result.outputs.out, sdk_type=Types.String)
```

Comment:
Shouldn’t this be list of floats

Reply:
https://github.com/lyft/flytesnacks/pull/11/files#diff-daa70d7f0f13b99d1e904cc0cb7fd7a7R152 I'm stringifying it here to be sure that UI will just have to print string I've provided.

Reply:
so we dont need to do that, since it is just an array of floats, UI can show it, if we mark it as an array of floats. Do you want to change that?